nodejs/device/src/logger.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+import { SeverityLevel } from "applicationinsights/out/Declarations/Contracts";
+import TelemetryClient from "applicationinsights/out/Library/TelemetryClient";
+import { format } from "util";
+
+export const enum LoggerSeverityLevel {
+  INFO = 1,
+  WARN = 2,
+  ERROR = 3,
+  CRITICAL = 4,
+}
+
+export type LoggerSettings = {
+  consoleEnabled: boolean;
+  consoleSeverityLevel?: LoggerSeverityLevel;
+  appInsightsEnabled: boolean;
+  appInsightsSeverityLevel?: LoggerSeverityLevel;
+  appInsightsClient?: TelemetryClient;
+};
+
+export class Logger {
+  constructor(settings: LoggerSettings) {
+    if (settings.consoleEnabled) {
+      if (!settings.consoleSeverityLevel) {
+        throw new Error(
+          "consoleSeverityLevel must be set if console logging is enabled."
+        );
+      }
+      this.consoleEnabled = true;
+      this.consoleSeverityLevel = settings.consoleSeverityLevel;
+    }
+    if (settings.appInsightsEnabled) {
+      if (!settings.appInsightsSeverityLevel || !settings.appInsightsClient) {
+        throw new Error(
+          "appInsightsSeverityLevel and appInsightsClient must be set if Application Insights logging is enabled."
+        );
+      }
+      this.appInsightsEnabled = true;
+      this.appInsightsSeverityLevel = settings.appInsightsSeverityLevel;
+      this.appInsightsClient = settings.appInsightsClient;
+    }
+  }
+
+  info = this.log.bind(this, LoggerSeverityLevel.INFO);
+  warn = this.log.bind(this, LoggerSeverityLevel.WARN);
+  error = this.log.bind(this, LoggerSeverityLevel.ERROR);
+  critical = this.log.bind(this, LoggerSeverityLevel.CRITICAL);
```

Comment:
These belong in the constructor -- and that's where tsc puts them anyway.  (except in the constructor, you need to specify this.info, etc instead of letting it be implied)

Reply:
In typescript the class members need to be declared. So you're saying that they should be declared without assignment and then assigned in the constructor?

Reply:
No, I was wrong.  I was translating to JS in my head and forgot about needing to declare members for type safety.